aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -319,6 +319,19 @@ impl<T: Transaction> CapturedReads<T> {
             .filter(|(_, v)| matches!(v, DataRead::Versioned(_, _, Some(_))))
     }
 
+    // Return an iterator over the captured group reads
+    // that contain a delayed field
+    pub(crate) fn get_group_read_values_with_delayed_fields(
+        &self,
+    ) -> impl Iterator<Item = (&T::Key, &GroupRead<T>)> {
+        self.group_reads.iter().filter(|(_, group_read)| {
+            group_read
```

Comment:
I am doing this filtering in `get_group_reads_needing_exchange_parallel` function. This function only returns the resources that have a layout and contains a delayed field aggregator that has been modified in this transaction.

Reply:
Adding a TODO here as well.

Reply:
The filtering uses extract_identfiers function which is part of latest_view. Unfortunately, it can't be implemented in captured reads.

Reply:
I still do not like returning GroupRead here, let's do some processing and return whatever is better for the caller already from this method (key, vec(tag, bytes, layout)), - if becomes a new aliased type, so be it - could also resolve the above todo and include ones with layouts only with layout (without option).

Also the caller clones things immediately and applies skip. We can pass skip logic here, and return a different data-structure if caller anyway clones and constructs some data-structure.

Reply:
@vusirikala @igor-aptos does the above make sense / is possible? I'm also happy if we do that refactor later, although it might not be big change and simplify things (if it could work as I imagine)?